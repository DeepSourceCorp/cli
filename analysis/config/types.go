package config

import "github.com/deepsourcelabs/cli/analysis/lsp"

///////////////////////////////////////////////////////
// DSConfig is the struct for .deepsource.toml file //
/////////////////////////////////////////////////////

type DSConfig struct {
	Version         int           `mapstructure:"version,omitempty" json:"version" toml:"version"`
	ExcludePatterns []string      `mapstructure:"exclude_patterns,omitempty" json:"exclude_patterns,omitempty" toml:"exclude_patterns"`
	TestPatterns    []string      `mapstructure:"test_patterns,omitempty" json:"test_patterns,omitempty" toml:"test_patterns"`
	Analyzers       []Analyzer    `mapstructure:"analyzers,omitempty" json:"analyzers,omitempty" toml:"analyzers"`
	Transformers    []Transformer `mapstructure:"transformers,omitempty" json:"transformers,omitempty" toml:"transformers"`
}

type Analyzer struct {
	Name                string      `mapstructure:"name,omitempty" json:"name,omitempty" toml:"name"`
	RuntimeVersion      string      `mapstructure:"runtime_version,omitempty" json:"runtime_version,omitempty" toml:"runtime_version,omitempty"`
	Enabled             bool        `mapstructure:"enabled,omitempty" json:"enabled" toml:"enabled"`
	DependencyFilePaths []string    `mapstructure:"dependency_file_paths,omitempty" json:"dependency_file_paths,omitempty"`
	Meta                interface{} `mapstructure:"meta,omitempty" json:"meta,omitempty" toml:"meta"`
	Thresholds          interface{} `mapstructure:"thresholds,omitempty" json:"thresholds,omitempty"`
}

type Transformer struct {
	Name    string `mapstructure:"name,omitempty" json:"name,omitempty" toml:"name"`
	Enabled bool   `mapstructure:"enabled,omitempty" json:"enabled,omitempty" toml:"enabled"`
}

///////////////////////////////////////
// LSP based Analysis Config Types  //
/////////////////////////////////////

type AnalysisConfig struct {
	Files           []lsp.TextDocumentItem `json:"files"`
	TestFiles       []lsp.TextDocumentItem `json:"test_files"`
	ExcludedFiles   []lsp.TextDocumentItem `json:"excluded_files"`
	ExcludePatterns []string               `json:"exclude_patterns"`
	TestPatterns    []string               `json:"test_patterns"`
	AnalyzerMeta    interface{}            `json:"analyzer_meta"`
}

//////////////////////////////////////
// LSP based Analysis Result Types  //
/////////////////////////////////////

type Namespace struct {
	Key   string  `json:"key"`
	Value float64 `json:"value"`
}

type Metric struct {
	MetricCode string      `json:"metric_code"`
	Namespaces []Namespace `json:"namespaces"`
}

type AnalysisResult struct {
	Issues   []lsp.Diagnostic `json:"issues"`
	Metrics  []Metric         `json:"metrics,omitempty"`
	IsPassed bool             `json:"is_passed"`
	Errors   []Error          `json:"errors"`
	// Errors    []lsp.Diagnostic `json:"errors"`
	ExtraData interface{} `json:"extra_data"`
}

////////////////////////////////////
// Default Analysis Result Types  //
///////////////////////////////////

type Error struct {
	HMessage string `json:"hmessage"`
	Level    int    `json:"level"`
}

type Coordinate struct {
	Line   int `json:"line"`
	Column int `json:"column"`
}

type Position struct {
	Begin Coordinate `json:"begin"`
	End   Coordinate `json:"end"`
}

type Location struct {
	Path     string   `json:"path"`
	Position Position `json:"position"`
}

type Issue struct {
	Code     string   `json:"issue_code"`
	Title    string   `json:"issue_text"`
	Location Location `json:"location"`
}

type DefaultAnalysisResult struct {
	Issues    []Issue     `json:"issues"`
	Metrics   []Metric    `json:"metrics,omitempty"`
	IsPassed  bool        `json:"is_passed"`
	Errors    []Error     `json:"errors"`
	ExtraData interface{} `json:"extra_data"`
}
