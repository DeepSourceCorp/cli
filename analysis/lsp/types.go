package lsp

//////////////////////////////
// Document identity types  //
/////////////////////////////

type DocumentURI string

type TextDocumentItem struct {
	URI        DocumentURI `json:"uri"`
	LanguageID string      `json:"languageID,omitempty"`
	Version    int         `json:"version,omitempty"`
	Text       string      `json:"text,omitempty"`
}

///////////////////////////////
// Diagnostic related types //
/////////////////////////////

type DiagnosticSeverity int

const (
	Error       DiagnosticSeverity = 1
	Warning     DiagnosticSeverity = 2
	Information DiagnosticSeverity = 3
	Hint        DiagnosticSeverity = 4
)

type Position struct {
	Line      int `json:"line"`
	Character int `json:"character"`
}
type Range struct {
	Start Position `json:"start"`
	End   Position `json:"end"`
}

type Location struct {
	URI   string `json:"uri"`
	Range Range  `json:"range"`
}

type DiagnosticRelatedInformation struct {
	Location Location `json:"location"`
	Message  string   `json:"message"`
}

type Diagnostic struct {
	Range    Range              `json:"range"`
	Severity DiagnosticSeverity `json:"severity,omitempty"`
	Code     string             `json:"code,omitempty"`
	Source   string             `json:"source,omitempty"`
	Message  string             `json:"message"`

	/**
	 * An array of related diagnostic information, e.g. when symbol-names within
	 * a scope collide all definitions can be marked via this property.
	 * var a,b
	 * a := 2
	 * Issues in line 1 and 2 are related.
	 */
	RelatedInformation []DiagnosticRelatedInformation `json:"relatedInformation"`
}
