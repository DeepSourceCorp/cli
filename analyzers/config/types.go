package config

type Analysis struct {
	Command string `toml:"command" json:"command" validate:"required"`
}

// TODO: Remove the comment when we start support for Autofix.
// type Autofix struct {
//     Command string `toml:"command" json:"command" analyzertoml:"required"`
// }

type Build struct {
	Builder    string `toml:"builder" json:"builder" validate:"omitempty"`
	Dockerfile string `toml:"dockerfile" json:"dockerfile,omitempty" validate:"omitempty"`
	Script     string `toml:"script" json:"script" validate:"omitempty"`
}

type URL struct {
	Source        string `toml:"source" json:"source" validate:"omitempty,url"`
	Documentation string `toml:"documentation" json:"documentation,omitempty" validate:"omitempty,url"`
	BugTracker    string `toml:"bug_tracker" json:"bug_tracker" validate:"omitempty,url"`
}

type Test struct {
	Command string `toml:"command" json:"command" validate:"omitempty"`
}

// analyzer.toml type
type AnalyzerMetadata struct {
	// Analyzer specific data
	Name                 string            `toml:"name" json:"name" validate:"required"`
	Shortcode            string            `toml:"shortcode" json:"shortcode" validate:"required"` // New
	Description          string            `toml:"description" json:"description" validate:"required"`
	Category             string            `toml:"category" json:"category" validate:"required"`
	Tags                 []string          `toml:"tags" json:"tags,omitempty" validate:"omitempty"`
	URL                  URL               `toml:"urls" json:"urls,omitempty" validate:"omitempty"`
	EnvironmentVariables map[string]string `toml:"environment_variables" json:"environment_variables,omitempty" validate:"omitempty"`

	// Analyzer, Autofix and Transformer config
	Analysis Analysis `toml:"analysis" json:"analysis" validate:"required"`
	// Autofix Autofix `toml:"autofix" json:"autofix" validate:"omitempty"`

	// Build steps needed to build the analyzer
	Build Build `toml:"build" json:"build" validate:"omitempty"`

	// Test command for running the testing workflow
	Test Test `toml:"test" json:"test" validate:"omitempty"`
}

// Analyzer issue description
type AnalyzerIssue struct {
	Shortcode   string `validate:"omitempty"`
	Title       string `toml:"title" json:"title" validate:"required"`
	Description string `toml:"description" json:"description" validate:"required"`
	Category    string `toml:"category" json:"category" validate:"required"`
}
