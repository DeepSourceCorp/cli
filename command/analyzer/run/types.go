package run

type AnalysisConfig struct {
	Files           []string    `json:"files"`
	ExcludePatterns []string    `json:"exclude_patterns"`
	ExcludeFiles    []string    `json:"exclude_files"`
	TestFiles       []string    `json:"test_files"`
	TestPatterns    []string    `json:"test_patterns"`
	AnalyzerMeta    interface{} `json:"analyzer_meta"`
}
