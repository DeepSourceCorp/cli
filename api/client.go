package api

import (
	"net"
	"net/http"
	"time"

	"github.com/shurcooL/graphql"
)

const DefaultReadTimeout = 10
const DefaultConnectTimeout = 1

var client *DSGQLClient

type DSQLClientConfig struct {
	ReadTimeout int
	ConnTimeout int
}

type DSGQLClient struct {
	Client *graphql.Client
	Token  string
}

func NewDSGQLClient(host string, token string, config *DSQLClientConfig) *DSGQLClient {
	httpClient := getHTTPClient(config.ConnTimeout, config.ReadTimeout)
	graqphqlClient := graphql.NewClient(host, httpClient)
	return &DSGQLClient{
		Client: graqphqlClient,
		Token:  token,
	}
}

func GetClient(hostname string, token string) *DSGQLClient {
	return NewDSGQLClient(hostname, token, &DSQLClientConfig{ReadTimeout: DefaultReadTimeout, ConnTimeout: DefaultConnectTimeout})
}

func getHTTPClient(connTimeOut int, readTimeout int) *http.Client {

	var netTransport = &http.Transport{
		Dial: (&net.Dialer{
			Timeout: time.Duration(connTimeOut) * time.Second,
		}).Dial,
		TLSHandshakeTimeout: 5 * time.Second,
	}

	var netClient = &http.Client{
		Timeout:   time.Duration(readTimeout) * time.Second,
		Transport: netTransport,
	}

	return netClient
}

// // To use it - gql := graphQLClient("host",client.http)
// func GraphQLClient(host string) *graphql.Client {
//     return graphql.NewClient(host, nil)
// }
