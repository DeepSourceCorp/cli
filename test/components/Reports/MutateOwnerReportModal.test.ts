import { render } from '@testing-library/vue'
import { createLocalVue, mount } from '@vue/test-utils'
import { VueConstructor } from 'vue'

import { MutateOwnerReportModal } from '~/components/Reports'
import CopyButton from '~/components/CopyButton.vue'

import VTooltip from 'v-tooltip'
import { focusDirective } from '~/plugins/helpers/directives.client'

import { ReportLevel, ReportSource } from '~/types/types'
import { ReportPageT } from '~/types/reportTypes'

interface MutateOwnerReportModalT extends Vue {
  //  Data properties
  isRestricted: boolean
  isPasswordHidden: boolean
  password: string
  reportKeys: Array<ReportPageT>
  shareHistoricalData: boolean
  reportLabel: string
  showPasswordInput: boolean
  showRepoSelection: boolean
  reportSource: ReportSource
  sourcedRepositories: Array<string>

  // Computed properties
  reportUrl: string

  // Methods
  handlePasswordProtectedToggle: (passworProtected: boolean) => void
  generatePassword: () => void
  togglePasswordVisibility: () => void
  handleResetPassword: () => void
  handleSourceUpdate: () => void
  handleSave: (close?: () => void) => void
  emitCreateReport: (close?: () => void) => void
  emitEditReport: (close?: () => void) => void
  validateArgs: () => boolean
  closeModal: () => void
}

describe('[[ MutateOwnerReportModal ]]', () => {
  let localVue: VueConstructor<Vue>

  const stubs = {
    ZModal: true,
    ZInput: true,
    ZButton: true,
    ZAlert: true,
    ZSelect: true,
    ZOption: true,
    ZIcon: true,
    ZToggle: true,
    ZCheckbox: true,
    ToggleInput: true,
    CopyButton: CopyButton
  }

  const mocks = {
    $route: {
      params: {
        provider: 'gh',
        owner: 'deepsourcelabs',
        repo: 'asgard'
      }
    },
    viewer: {},
    $config: { onPrem: false, domain: 'deepsource.io' },
    $socket: {
      $on: () => {},
      $off: () => {}
    },
    $toast: {
      danger: jest.fn(),
      success: jest.fn()
    },
    async $fetchGraphqlData() {},
    $providerMetaMap: {
      gh: {
        text: 'Github',
        shortcode: 'gh',
        value: 'GITHUB'
      }
    }
  }

  beforeEach(() => {
    localVue = createLocalVue()
    localVue.use(VTooltip)
    localVue.directive('focus', focusDirective)
  })

  test('renders MutateOwnerReportModal in create report mode', () => {
    const props = {
      editMode: false,
      level: ReportLevel.Repository
    }

    const { html } = render(MutateOwnerReportModal, {
      props,
      stubs,
      localVue
    })

    expect(html()).toMatchSnapshot(JSON.stringify(props))
  })

  test('renders MutateOwnerReportModal in edit report mode', () => {
    const props = {
      editMode: false,
      level: ReportLevel.Repository,
      isRestrictedOld: true,
      reportIdOld: 'aa183352-4b5e-427a-ac8d-95c47c52b6a1',
      reportKeysOld: [ReportPageT.OWASP_TOP_10, ReportPageT.SANS_TOP_25],
      reportLabelOld: 'Sample Report',
      shareHistoricalDataOld: true,
      reportSourceOld: ReportSource.SourceSelected,
      sourcedRepositoriesOld: [
        'UmVwb3NpdG9yeTp6dmp2eXo=',
        'UmVwb3NpdG9yeTp6dmpleXo=',
        'UmVwb3NpdG9yeTp6Z2tlZHo='
      ]
    }

    const { html } = render(MutateOwnerReportModal, {
      props,
      stubs,
      mocks,
      localVue
    })

    expect(html()).toMatchSnapshot(JSON.stringify(props))
  })

  test('MutateOwnerReportModal computed properties and methods', () => {
    const propsData = {
      editMode: true,
      level: ReportLevel.Repository,
      isRestrictedOld: false,
      reportIdOld: 'aa183352-4b5e-427a-ac8d-95c47c52b6a1',
      reportKeysOld: [ReportPageT.OWASP_TOP_10, ReportPageT.SANS_TOP_25],
      reportLabelOld: 'Sample Report',
      shareHistoricalDataOld: true,
      reportSourceOld: ReportSource.SourceAll
    }
    const wrapper = mount(MutateOwnerReportModal, {
      stubs,
      propsData,
      mocks,
      localVue
    })

    const vm = wrapper.vm as MutateOwnerReportModalT

    // ------ Test computed property ------
    expect(vm.reportUrl).toBe('https://deepsource.io/report/aa183352-4b5e-427a-ac8d-95c47c52b6a1')

    // ------ Test methods ------

    // Test generatePassword sets password data property
    expect(vm.password).toBe('')
    vm.generatePassword()
    expect(vm.password.length).toBe(16)

    // Test generatePassword generates different passwords on each call
    vm.generatePassword()
    const firstPassword = vm.password
    vm.generatePassword()
    const secondPassword = vm.password
    expect(firstPassword).not.toBe(secondPassword)

    // Test togglePasswordVisibility
    expect(vm.isPasswordHidden).toBe(false)
    vm.togglePasswordVisibility()
    expect(vm.isPasswordHidden).toBe(true)

    // Test handleResetPassword
    vm.password = ''
    vm.handleResetPassword()
    expect(vm.password.length).toBe(16)
    expect(vm.showPasswordInput).toBe(true)

    // Test handlePasswordProtectedToggle for passwordProtected = true
    vm.password = ''
    vm.showPasswordInput = false
    vm.handlePasswordProtectedToggle(true)
    expect(vm.password.length).toBe(16)
    expect(vm.showPasswordInput).toBe(true)

    // Test handlePasswordProtectedToggle for passwordProtected = false
    vm.password = 'sample-password'
    vm.showPasswordInput = true
    vm.handlePasswordProtectedToggle(false)
    expect(vm.password).toBe('')
    expect(vm.showPasswordInput).toBe(false)

    // Test handleSourceUpdate
    vm.reportSource = ReportSource.SourceSelected
    vm.handleSourceUpdate()
    expect(vm.showRepoSelection).toBe(true)

    // Test closeModal
    vm.closeModal()
    expect(wrapper.emitted()).toHaveProperty('close')

    // ----- Test validateArgs -----

    // Test error for only empty label
    vm.reportLabel = ''
    vm.isRestricted = false
    vm.reportKeys = [ReportPageT.OWASP_TOP_10]
    expect(vm.validateArgs()).toBe(false)

    // Test error for only no reports selected
    vm.reportLabel = 'Sample Report'
    vm.isRestricted = false
    vm.reportKeys = []
    expect(vm.validateArgs()).toBe(false)

    // Test error for only password
    vm.reportLabel = 'Sample Report'
    vm.reportKeys = [ReportPageT.OWASP_TOP_10]
    vm.showPasswordInput = true
    vm.password = ''
    expect(vm.validateArgs()).toBe(false)

    // Test error for reportSource
    vm.reportLabel = 'Sample Report'
    vm.showPasswordInput = false
    vm.reportKeys = [ReportPageT.OWASP_TOP_10]
    vm.reportSource = ReportSource.SourceSelected
    vm.sourcedRepositories = []
    expect(vm.validateArgs()).toBe(false)

    // Test validateArgs for all correct args
    vm.reportLabel = 'Sample Report'
    vm.reportKeys = [ReportPageT.SANS_TOP_25, ReportPageT.DISTRIBUTION]
    vm.showPasswordInput = true
    vm.password = 'sample-password'
    vm.reportSource = ReportSource.SourceSelected
    vm.sourcedRepositories = [
      'UmVwb3NpdG9yeTp6dmp2eXo=',
      'UmVwb3NpdG9yeTp6dmpleXo=',
      'UmVwb3NpdG9yeTp6Z2tlZHo='
    ]

    expect(vm.validateArgs()).toBe(true)
  })

  test('Report creation flow', async () => {
    const propsData = {
      editMode: false,
      level: ReportLevel.Repository
    }
    const wrapper = await mount(MutateOwnerReportModal, {
      stubs: { ...stubs, ZModal: false },
      propsData,
      mocks,
      localVue
    })

    const vm = wrapper.vm as MutateOwnerReportModalT

    vm.reportLabel = 'Sample Report'
    vm.reportKeys = [ReportPageT.OWASP_TOP_10, ReportPageT.SANS_TOP_25]
    vm.isRestricted = true
    vm.password = 'sample-password'
    vm.shareHistoricalData = true
    vm.reportSource = ReportSource.SourceSelected
    vm.sourcedRepositories = [
      'UmVwb3NpdG9yeTp6dmp2eXo=',
      'UmVwb3NpdG9yeTp6dmpleXo=',
      'UmVwb3NpdG9yeTp6Z2tlZHo='
    ]

    vm.handleSave()

    // Assertions
    expect(wrapper.emitted()).toHaveProperty('create-report')
    expect(wrapper.emitted()['create-report']).toStrictEqual([
      [
        {
          ownerLogin: 'deepsourcelabs',
          repositoryName: 'asgard',
          vcsProvider: vm.$providerMetaMap['gh'].value,
          label: 'Sample Report',
          level: ReportLevel.Repository,
          isRestricted: true,
          password: 'sample-password',
          shareHistoricalData: true,
          reportKeys: [ReportPageT.OWASP_TOP_10, ReportPageT.SANS_TOP_25],
          source: ReportSource.SourceSelected,
          sourcedRepositories: [
            'UmVwb3NpdG9yeTp6dmp2eXo=',
            'UmVwb3NpdG9yeTp6dmpleXo=',
            'UmVwb3NpdG9yeTp6Z2tlZHo='
          ]
        },
        undefined
      ]
    ])
  })

  test('Report updation flow', async () => {
    const propsData = {
      editMode: true,
      level: ReportLevel.Repository,
      isRestrictedOld: false,
      reportIdOld: 'aa183352-4b5e-427a-ac8d-95c47c52b6a1',
      reportKeysOld: [ReportPageT.OWASP_TOP_10, ReportPageT.SANS_TOP_25],
      reportLabelOld: 'Sample Report',
      shareHistoricalDataOld: false,
      reportSourceOld: ReportSource.SourceAll
    }
    const wrapper = await mount(MutateOwnerReportModal, {
      stubs: { ...stubs, ZModal: false },
      propsData,
      mocks,
      localVue
    })

    const vm = wrapper.vm as MutateOwnerReportModalT

    vm.reportLabel = 'Updated Report'
    vm.reportKeys = [ReportPageT.OWASP_TOP_10, ReportPageT.DISTRIBUTION]
    vm.isRestricted = true
    vm.password = 'sample-password'
    vm.shareHistoricalData = true
    vm.reportSource = ReportSource.SourceSelected
    vm.sourcedRepositories = [
      'UmVwb3NpdG9yeTp6dmp2eXo=',
      'UmVwb3NpdG9yeTp6dmpleXo=',
      'UmVwb3NpdG9yeTp6Z2tlZHo='
    ]

    vm.handleSave()

    // Assertions
    expect(wrapper.emitted()).toHaveProperty('edit-report')
    expect(wrapper.emitted()['edit-report']).toStrictEqual([
      [
        {
          reportId: 'aa183352-4b5e-427a-ac8d-95c47c52b6a1',
          label: 'Updated Report',
          shareHistoricalData: true,
          reportKeys: [ReportPageT.OWASP_TOP_10, ReportPageT.DISTRIBUTION],
          isRestricted: true,
          password: 'sample-password',
          source: ReportSource.SourceSelected,
          sourcedRepositories: [
            'UmVwb3NpdG9yeTp6dmp2eXo=',
            'UmVwb3NpdG9yeTp6dmpleXo=',
            'UmVwb3NpdG9yeTp6Z2tlZHo='
          ]
        },
        undefined
      ]
    ])
  })
})

describe('[[ Test setTimeout for reseting password visibility ]]', () => {
  let localVue: VueConstructor<Vue>

  const stubs = {
    ZModal: true,
    ZInput: true,
    ZButton: true,
    ZAlert: true,
    ZSelect: true,
    ZOption: true,
    ZIcon: true,
    ZToggle: true,
    ZCheckbox: true,
    ToggleInput: true,
    CopyButton: CopyButton
  }

  const mocks = {
    $route: {
      params: {
        provider: 'gh',
        owner: 'deepsourcelabs',
        repo: 'asgard'
      }
    },
    viewer: {},
    $config: { onPrem: false, domain: 'deepsource.io' },
    $socket: {
      $on: () => {},
      $off: () => {}
    },
    $toast: {
      danger: jest.fn(),
      success: jest.fn()
    },
    async $fetchGraphqlData() {},
    $providerMetaMap: {
      gh: {
        text: 'Github',
        shortcode: 'gh',
        value: 'GITHUB'
      }
    }
  }

  beforeEach(() => {
    localVue = createLocalVue()
    localVue.use(VTooltip)
    localVue.directive('focus', focusDirective)
    jest.useFakeTimers()
  })

  afterEach(() => {
    jest.useRealTimers()
  })

  test('Password hide setTimeout', async () => {
    const propsData = {
      editMode: false,
      level: ReportLevel.Repository
    }
    const wrapper = await mount(MutateOwnerReportModal, {
      stubs: { ...stubs, ZModal: false },
      propsData,
      mocks,
      localVue
    })

    const vm = wrapper.vm as MutateOwnerReportModalT

    vm.isPasswordHidden = true

    vm.generatePassword()

    expect(vm.isPasswordHidden).toBe(false)
    jest.runAllTimers()
    expect(vm.isPasswordHidden).toBe(true)
  })
})
