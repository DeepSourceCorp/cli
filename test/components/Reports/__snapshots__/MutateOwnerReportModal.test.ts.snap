// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[[ render MutateOwnerReportModal ]] renders MutateOwnerReportModal in create report mode: {"level":"OWNER","ownerLogin":"deepsourcelabs","vcsProvider":"gh","editMode":false} 1`] = `
"<zmodal-stub width=\\"base\\" closeafterprimaryaction=\\"true\\" primaryactiontype=\\"primary\\" showheaderborder=\\"true\\">
  <fieldset class=\\"p-4 flex flex-col gap-y-7\\"><label for=\\"report-title\\" class=\\"space-y-2 text-sm text-vanilla-200 w-full\\">
      <p class=\\"text-sm font-medium text-vanilla-100\\">Label</p>
      <zinput-stub name=\\"\\" value=\\"\\" label=\\"\\" type=\\"text\\" placeholder=\\"What’s this report for?\\" debouncedelay=\\"350\\" size=\\"small\\" backgroundcolor=\\"ink-400\\" errorbordercolor=\\"cherry\\" required=\\"true\\" id=\\"report-title\\"></zinput-stub>
      <p class=\\"flex items-center gap-x-1 text-vanilla-400 text-xs\\">
        This is used for internal reference only
      </p>
    </label>
    <!---->
    <div class=\\"space-y-3\\">
      <toggleinput-stub input-width=\\"xx-small\\" label=\\"Restrict report access\\" input-id=\\"toggle-password-protection\\" remove-y-padding=\\"true\\" class=\\"font-medium\\"><template></template></toggleinput-stub>
      <!---->
    </div>
    <div class=\\"grid grid-cols-2\\">
      <div class=\\"flex flex-col gap-y-2\\">
        <p class=\\"text-xs uppercase text-vanilla-400 font-medium tracking-wider\\">Security</p>
        <zcheckbox-stub label=\\"OWASP Top 10\\" name=\\"\\" value=\\"owasp-top-10\\" truevalue=\\"true\\" modelvalue=\\"\\" size=\\"small\\" class=\\"cursor-pointer\\"></zcheckbox-stub>
        <zcheckbox-stub label=\\"SANS Top 25\\" name=\\"\\" value=\\"sans-top-25\\" truevalue=\\"true\\" modelvalue=\\"\\" size=\\"small\\" class=\\"cursor-pointer\\"></zcheckbox-stub>
        <!---->
        <!---->
        <!---->
        <!---->
        <!---->
        <!---->
      </div>
      <div class=\\"flex flex-col gap-y-2\\">
        <p class=\\"text-xs uppercase text-vanilla-400 font-medium tracking-wider\\">Insights</p>
        <!---->
        <!---->
        <zcheckbox-stub label=\\"Code Health Trend\\" name=\\"\\" value=\\"code-health-trend\\" truevalue=\\"true\\" modelvalue=\\"\\" size=\\"small\\" class=\\"cursor-pointer\\"></zcheckbox-stub>
        <zcheckbox-stub label=\\"Issue Distribution\\" name=\\"\\" value=\\"issue-distribution\\" truevalue=\\"true\\" modelvalue=\\"\\" size=\\"small\\" class=\\"cursor-pointer\\"></zcheckbox-stub>
        <zcheckbox-stub label=\\"Code Coverage\\" name=\\"\\" value=\\"code-coverage\\" truevalue=\\"true\\" modelvalue=\\"\\" size=\\"small\\" class=\\"cursor-pointer\\"></zcheckbox-stub>
        <zcheckbox-stub label=\\"Issues Prevented\\" name=\\"\\" value=\\"issues-prevented\\" truevalue=\\"true\\" modelvalue=\\"\\" size=\\"small\\" class=\\"cursor-pointer\\"></zcheckbox-stub>
        <zcheckbox-stub label=\\"Issues Autofixed\\" name=\\"\\" value=\\"issues-autofixed\\" truevalue=\\"true\\" modelvalue=\\"\\" size=\\"small\\" class=\\"cursor-pointer\\"></zcheckbox-stub>
        <!---->
      </div>
    </div>
    <toggleinput-stub input-width=\\"xx-small\\" label=\\"Share historical data\\" input-id=\\"toggle-share-historical-data\\" remove-y-padding=\\"true\\" class=\\"font-medium\\"><template></template></toggleinput-stub> <label for=\\"source-repos\\" class=\\"space-y-2 text-sm\\">
      <p class=\\"font-medium text-vanilla-100\\">Select repositories</p>
      <zradiogroup-stub modelvalue=\\"SOURCE_ALL\\" id=\\"source-repos\\" class=\\"grid grid-cols-2\\">
        <zradio-stub modelvalue=\\"\\" label=\\"All repositories\\" value=\\"SOURCE_ALL\\"></zradio-stub>
        <zradio-stub modelvalue=\\"\\" label=\\"Custom\\" value=\\"SOURCE_SELECTED\\"></zradio-stub>
      </zradiogroup-stub>
    </label>
  </fieldset>
  <selectrepositoriesforreport-stub report-id=\\"\\" owner-login=\\"deepsourcelabs\\" vcs-provider=\\"gh\\" style=\\"display: none;\\"></selectrepositoriesforreport-stub>
</zmodal-stub>"
`;

exports[`[[ render MutateOwnerReportModal ]] renders MutateOwnerReportModal in edit report mode: {"level":"OWNER","ownerLogin":"deepsourcelabs","vcsProvider":"gh","editMode":false,"isRestrictedOld":true,"reportIdOld":"aa183352-4b5e-427a-ac8d-95c47c52b6a1","reportKeysOld":["owasp-top-10","sans-top-25"],"reportLabelOld":"Sample Report","shareHistoricalDataOld":true,"reportSourceOld":"SOURCE_SELECTED","sourceRepoCountOld":2} 1`] = `
"<zmodal-stub width=\\"base\\" closeafterprimaryaction=\\"true\\" primaryactiontype=\\"primary\\" showheaderborder=\\"true\\">
  <fieldset class=\\"p-4 flex flex-col gap-y-7\\"><label for=\\"report-title\\" class=\\"space-y-2 text-sm text-vanilla-200 w-full\\">
      <p class=\\"text-sm font-medium text-vanilla-100\\">Label</p>
      <zinput-stub name=\\"\\" value=\\"\\" label=\\"\\" type=\\"text\\" placeholder=\\"What’s this report for?\\" debouncedelay=\\"350\\" size=\\"small\\" backgroundcolor=\\"ink-400\\" errorbordercolor=\\"cherry\\" required=\\"true\\" id=\\"report-title\\"></zinput-stub>
      <p class=\\"flex items-center gap-x-1 text-vanilla-400 text-xs\\">
        This is used for internal reference only
      </p>
    </label> <label for=\\"report-link\\" class=\\"space-y-2 text-sm text-vanilla-200 w-full\\">
      <p class=\\"text-sm font-medium text-vanilla-100\\">Shareable link</p>
      <zinput-stub name=\\"\\" value=\\"https://deepsource.io/report/aa183352-4b5e-427a-ac8d-95c47c52b6a1\\" label=\\"\\" type=\\"text\\" placeholder=\\"Shareable link for the report...\\" readonly=\\"true\\" debouncedelay=\\"350\\" size=\\"medium\\" backgroundcolor=\\"ink-400\\" errorbordercolor=\\"cherry\\" id=\\"report-link\\"><template></template></zinput-stub>
    </label>
    <div class=\\"space-y-3\\">
      <toggleinput-stub input-width=\\"xx-small\\" label=\\"Restrict report access\\" input-id=\\"toggle-password-protection\\" remove-y-padding=\\"true\\" class=\\"font-medium\\"><template></template></toggleinput-stub>
      <!---->
    </div>
    <div class=\\"grid grid-cols-2\\">
      <div class=\\"flex flex-col gap-y-2\\">
        <p class=\\"text-xs uppercase text-vanilla-400 font-medium tracking-wider\\">Security</p>
        <zcheckbox-stub label=\\"OWASP Top 10\\" name=\\"\\" value=\\"owasp-top-10\\" truevalue=\\"true\\" modelvalue=\\"\\" size=\\"small\\" class=\\"cursor-pointer\\"></zcheckbox-stub>
        <zcheckbox-stub label=\\"SANS Top 25\\" name=\\"\\" value=\\"sans-top-25\\" truevalue=\\"true\\" modelvalue=\\"\\" size=\\"small\\" class=\\"cursor-pointer\\"></zcheckbox-stub>
        <!---->
        <!---->
        <!---->
        <!---->
        <!---->
        <!---->
      </div>
      <div class=\\"flex flex-col gap-y-2\\">
        <p class=\\"text-xs uppercase text-vanilla-400 font-medium tracking-wider\\">Insights</p>
        <!---->
        <!---->
        <zcheckbox-stub label=\\"Code Health Trend\\" name=\\"\\" value=\\"code-health-trend\\" truevalue=\\"true\\" modelvalue=\\"\\" size=\\"small\\" class=\\"cursor-pointer\\"></zcheckbox-stub>
        <zcheckbox-stub label=\\"Issue Distribution\\" name=\\"\\" value=\\"issue-distribution\\" truevalue=\\"true\\" modelvalue=\\"\\" size=\\"small\\" class=\\"cursor-pointer\\"></zcheckbox-stub>
        <zcheckbox-stub label=\\"Code Coverage\\" name=\\"\\" value=\\"code-coverage\\" truevalue=\\"true\\" modelvalue=\\"\\" size=\\"small\\" class=\\"cursor-pointer\\"></zcheckbox-stub>
        <zcheckbox-stub label=\\"Issues Prevented\\" name=\\"\\" value=\\"issues-prevented\\" truevalue=\\"true\\" modelvalue=\\"\\" size=\\"small\\" class=\\"cursor-pointer\\"></zcheckbox-stub>
        <zcheckbox-stub label=\\"Issues Autofixed\\" name=\\"\\" value=\\"issues-autofixed\\" truevalue=\\"true\\" modelvalue=\\"\\" size=\\"small\\" class=\\"cursor-pointer\\"></zcheckbox-stub>
        <!---->
      </div>
    </div>
    <toggleinput-stub input-width=\\"xx-small\\" label=\\"Share historical data\\" input-id=\\"toggle-share-historical-data\\" remove-y-padding=\\"true\\" class=\\"font-medium\\"><template></template></toggleinput-stub> <label for=\\"source-repos\\" class=\\"space-y-2 text-sm\\">
      <p class=\\"font-medium text-vanilla-100\\">Select repositories</p>
      <zradiogroup-stub modelvalue=\\"SOURCE_ALL\\" id=\\"source-repos\\" class=\\"grid grid-cols-2\\">
        <zradio-stub modelvalue=\\"\\" label=\\"All repositories\\" value=\\"SOURCE_ALL\\"></zradio-stub>
        <zradio-stub modelvalue=\\"\\" label=\\"Custom\\" value=\\"SOURCE_SELECTED\\"></zradio-stub>
      </zradiogroup-stub>
    </label>
  </fieldset>
  <selectrepositoriesforreport-stub report-id=\\"aa183352-4b5e-427a-ac8d-95c47c52b6a1\\" owner-login=\\"deepsourcelabs\\" vcs-provider=\\"gh\\" style=\\"display: none;\\"></selectrepositoriesforreport-stub>
</zmodal-stub>"
`;
